disputer/disputer.js

```diff
@@ -46,18 +48,9 @@ class Disputer {
     };
 
     // Set and validate config settings.
-    // TODO: Refactor this functionality into a common/ module.
-    Object.keys(defaultConfig).forEach(field => {
-      this[field] = config && field in config ? config[field] : defaultConfig[field].value;
-      if (!defaultConfig[field].isValid(this[field])) {
-        this.logger.error({
-          at: "Disputer",
-          message: "Attempting to set configuration field with invalid value",
-          field: field,
-          value: this[field]
-        });
-        throw new Error("Attempting to set configuration field with invalid value");
-      }
+    const newConfig = createObjectFromDefaultProps(config, defaultConfig);
+    Object.keys(newConfig).forEach(setting => {
+      this[setting] = newConfig[setting];
```

Comment:
nit: I think you can do:
```
Object.assign(this, newConfig);
```
for a slightly simpler syntax.

Reply:
wow this is great

Reply:
Ended up making this a one-liner

`Object.assign(this, createObjectFromDefaultProps(config, defaultConfig));`